Core/Graphics/Source/Graphics.cpp

```diff
@@ -64,12 +64,23 @@ namespace Babylon
 
     void Graphics::Impl::Resize(size_t width, size_t height)
     {
-        std::scoped_lock lock{m_bgfxState.Mutex};
-        m_bgfxState.Dirty = true;
+        {
+            std::scoped_lock lock(m_resolution.Mutex);
```

Comment:
Is the reason `m_resolution` has its own mutex because it doesn't seem like a natural part of `m_bgfxState`? If so, it might be better to rename `m_bgfxState` to just `m_internalState`, or
```
struct
{
    std::mutex Mutex{};

    struct
    {
        ...
    } Resolution{};

    struct
    {
        ...
    } Bgfx{};
} m_state{};
```
or something like that, to avoid having to unlock and lock two different mutexes that are both trying to guard two very closely related parts of the Graphics object's state.

Reply:
That was my reasoning for the most part, this isn't quite appropriate to call this the state of BGFX, but some internal state that we keep and use to set BGFX's state. I think locking all of this behind a single mutex isn't a bad change, especially considering how a change to one affects the other.

Reply:
refactored the structs to a single struct following your suggestion, definitely makes the locking a bit more understandable.